### ISSUE TITLE
nodes fail to come all the way up after restart

### ISSUE THREAD

**Issue by snstanton** [bottlerocket-os/bottlerocket#1594]
The upgrader automatically upgraded our nodes this morning, which is great.  Unfortunately we're still seeing the behavior where there is a startup race with aws-node, so one of our nodes didn't properly restart.  I had to manually kill the aws-node pod on that node to get it to finish the upgrade.  We see this pretty much every time an upgrade happens.

There seems to be a race condition where aws-node starts up before it can connect to one of the services it depends on and gets stuck.  Killing the pod causes it to come up cleanly and everything else appears to be fine.

This is the output of the aws-node pod when the upgrade get stuck:
```
{"level":"info","ts":"2021-03-19T17:25:48.095Z","caller":"entrypoint.sh","msg":"Install CNI binary.."}
{"level":"info","ts":"2021-03-19T17:25:48.171Z","caller":"entrypoint.sh","msg":"Starting IPAM daemon in the background ... "}
{"level":"info","ts":"2021-03-19T17:25:48.172Z","caller":"entrypoint.sh","msg":"Checking for IPAM connectivity ... "}
ERROR: logging before flag.Parse: E0319 17:26:18.429644      10 memcache.go:138] couldn't get current server API group list; will keep using cached value. (Get https://10.100.0.1:443/api?timeout=32s: dial tcp 10.100.0.1:443: i/o timeout)
```

**Comment by jpculp** [comment]
Hi @snstanton, just wanted to touch base. Are you still needing to do the `--bind-address` workaround?

**Comment by snstanton** [comment]
We aren't currently adding that arg.  I do still see `--hostname-override=$(NODE_NAME)` in the args list, though.  I don't know if that's needed or not.

**Comment by etungsten** [comment]
According to https://github.com/kubernetes/kubernetes/issues/61486#issuecomment-1275961318, this issue has been fixed in EKS v1.22.6-eksbuild.1. And the workaround with adding `--hostname-override=$(NODE_NAME)` the `kube-proxy` invocation still works if people are still running into this issue. Closing this issue for now.
